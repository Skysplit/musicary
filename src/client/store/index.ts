import { createStore } from 'redux';
import { StoreCreatorOptions } from 'next-redux-wrapper';
import { Request } from 'express';
import rootReducer from './rootReducer';
import middleware from './middleware';

export type State = ReturnType<typeof rootReducer>;

interface Options extends StoreCreatorOptions<State, {}, {}, {}, {}> {
  req: Request;
}

export default (initialState: State, options: Options) => {
  return createStore(rootReducer, middleware);
};
