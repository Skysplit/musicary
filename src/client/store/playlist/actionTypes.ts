export const FETCH_MANY_PLAYLISTS_REQUEST = 'playlist/FETCH_MANY_REQUEST';
export const FETCH_MANY_PLAYLISTS_SUCCESS = 'playlist/FETCH_MANY_SUCCESS';
export const FETCH_MANY_PLAYLISTS_FAILURE = 'playlist/FETCH_MANY_FAILURE';

export const FETCH_SINGLE_PLAYLIST_REQUEST = 'playlist/FETCH_SINGLE_REQUEST';
export const FETCH_SINGLE_PLAYLIST_SUCCESS = 'playlist/FETCH_SINGLE_SUCCESS';
export const FETCH_SINGLE_PLAYLIST_FAILURE = 'playlist/FETCH_SINGLE_FAILURE';

export const REMOVE_PLAYLIST_REQUEST = 'playlist/REMOVE_REQUEST';
export const REMOVE_PLAYLIST_SUCCESS = 'playlist/REMOVE_SUCCESS';
export const REMOVE_PLAYLIST_FAILURE = 'playlist/REMOVE_FAILURE';

export const SET_PLAYLIST = 'playlist/SET';
