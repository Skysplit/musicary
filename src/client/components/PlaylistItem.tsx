import React, { PureComponent, MouseEvent } from 'react';
import { PlaylistInterface } from '@client/store/playlist';
import { Grid, Cell, Button } from 'react-md';
import Link from '@next/link';

interface PlaylistItemProps {
  playlist: PlaylistInterface;
  removing: boolean;
  onRemove: (playlist: PlaylistInterface) => any;
  onEdit: () => any;
}

export default class PlaylistItem extends PureComponent<PlaylistItemProps> {
  handleRemove = (e: MouseEvent<HTMLElement>) => {
    e.stopPropagation();

    const { onRemove, playlist } = this.props;
    const confirmed = confirm(`Are you sure you want to remove playlist ${playlist.name}?`);

    if (confirmed) {
      onRemove(this.props.playlist);
    }
  }

  handleEdit = (e: MouseEvent<HTMLElement>) => {
    e.stopPropagation();
    this.props.onEdit();
  }

  render() {
    const { playlist, removing } = this.props;

    return (
      <Link route="playlist" params={{ id: playlist.id }}>
        <Grid gutter={0} spacing={0} noSpacing style={{ cursor: 'pointer' }}>
          <Cell phoneSize={3} tabletSize={6} desktopSize={11} align="middle">
            <span className="md-title">
              {playlist.name}
            </span>
          </Cell>
          <Cell phoneSize={1} tabletSize={2} desktopSize={1} style={{ textAlign: 'right' }}>
            <Button
              className="edit"
              icon
              primary
              onClick={this.handleEdit}
              disabled={removing}
            >
              edit
            </Button>{' '}
            <Button
              icon
              secondary
              onClick={this.handleRemove}
              disabled={removing}
              className="remove"
            >
              delete
            </Button>
          </Cell>
        </Grid>
      </Link>
    );
  }
}
