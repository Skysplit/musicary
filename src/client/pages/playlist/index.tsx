import React, { PureComponent } from 'react';
import { Request } from 'express';
import { bindActionCreators } from 'redux';
import { connect, Dispatch } from 'react-redux';
import Error from 'next/error';
import { Context } from 'next/document';
import PlaylistsListContainer from '@client/containers/PlaylistsListContainer';
import client, { getHeaders } from '@client/utils/client';
import { PlaylistInterface } from '@client/store/playlist';
import { fetchPlaylistsSuccess } from '@client/store/playlist/actions';

interface PropsContext extends Context {
  req: Request;
}

interface InitialProps {
  title: string;
  errorCode?: number;
  playlists?: PlaylistInterface[];
}

interface PlaylistsPageProps extends InitialProps {
  fetchPlaylistsSuccess: typeof fetchPlaylistsSuccess;
}

export class PlaylistsPage extends PureComponent<PlaylistsPageProps> {
  static async fetchPlaylists(req: Request) {
    const headers = getHeaders(req);
    const response = await client.get('/playlists', { headers });
    return response.data as PlaylistInterface[];
  }

  static async getInitialProps(ctx: PropsContext): Promise<InitialProps> {
    let errorCode: number;
    let playlists: PlaylistInterface[];

    try {
      playlists = await PlaylistsPage.fetchPlaylists(ctx.req);
    } catch (err) {
      errorCode = err.response.status;
    }

    return {
      errorCode,
      playlists,
      title: 'My playlists',
    };
  }

  constructor(props: PlaylistsPageProps) {
    super(props);

    if (props.playlists) {
      props.fetchPlaylistsSuccess(props.playlists);
    }
  }

  render() {
    const { errorCode } = this.props;

    if (errorCode) {
      return <Error statusCode={errorCode} />;
    }

    return (
      <PlaylistsListContainer />
    );
  }
}

const actionCreators = {
  fetchPlaylistsSuccess,
};

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(actionCreators, dispatch);

export default connect(null, mapDispatchToProps)(PlaylistsPage);
