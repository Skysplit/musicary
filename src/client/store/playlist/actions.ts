import {
  FETCH_MANY_PLAYLISTS_REQUEST,
  FETCH_MANY_PLAYLISTS_SUCCESS,
  FETCH_SINGLE_PLAYLIST_REQUEST,
  FETCH_MANY_PLAYLISTS_FAILURE,
  FETCH_SINGLE_PLAYLIST_FAILURE,
  FETCH_SINGLE_PLAYLIST_SUCCESS,
  SET_PLAYLIST,
  REMOVE_PLAYLIST_REQUEST,
  REMOVE_PLAYLIST_FAILURE,
  REMOVE_PLAYLIST_SUCCESS,
  REMOVE_PLAYLIST_TRACK,
} from '@client/store/playlist/actionTypes';
import { PlaylistInterface } from '@client/store/playlist';

// Fetching many playlists
export const fetchPlaylistsRequest = () => ({
  type: FETCH_MANY_PLAYLISTS_REQUEST as typeof FETCH_MANY_PLAYLISTS_REQUEST,
});

export const fetchPlaylistsSuccess = (payload: PlaylistInterface[]) => ({
  payload,
  type: FETCH_MANY_PLAYLISTS_SUCCESS as typeof FETCH_MANY_PLAYLISTS_SUCCESS,
});

export const fetchPlaylistsFailure = () => ({
  type: FETCH_MANY_PLAYLISTS_FAILURE as typeof FETCH_MANY_PLAYLISTS_FAILURE,
});

// Fetching single playlist
export const fetchPlaylistRequest = (id: number) => ({
  payload: id,
  type: FETCH_SINGLE_PLAYLIST_REQUEST as typeof FETCH_SINGLE_PLAYLIST_REQUEST,
});

export const fetchPlaylistSuccess = (payload: PlaylistInterface) => ({
  payload,
  type: FETCH_SINGLE_PLAYLIST_SUCCESS as typeof FETCH_SINGLE_PLAYLIST_SUCCESS,
});

export const fetchPlaylistFailure = (id: number) => ({
  payload: id,
  type: FETCH_SINGLE_PLAYLIST_FAILURE as typeof FETCH_SINGLE_PLAYLIST_FAILURE,
});

// Delete single playlist
export const removePlaylistRequest = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_REQUEST as typeof REMOVE_PLAYLIST_REQUEST,
});

export const removePlaylistSuccess = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_SUCCESS as typeof REMOVE_PLAYLIST_SUCCESS,
});

export const removePlaylistFailure = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_FAILURE as typeof REMOVE_PLAYLIST_FAILURE,
});

// Save single playlist
export const setSinglePlaylist = (payload: PlaylistInterface) => ({
  payload,
  type: SET_PLAYLIST as typeof SET_PLAYLIST,
});

export const removePlaylistTrack = (playlistId: number, trackId: number) => ({
  type: REMOVE_PLAYLIST_TRACK as typeof REMOVE_PLAYLIST_TRACK,
  payload: {
    playlistId,
    trackId,
  },
});

