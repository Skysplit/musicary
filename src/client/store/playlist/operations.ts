import { Dispatch } from 'react-redux';
import {
  fetchPlaylistsRequest,
  fetchPlaylistsSuccess,
  fetchPlaylistsFailure,
  fetchPlaylistRequest,
  fetchPlaylistSuccess,
  fetchPlaylistFailure,
  setSinglePlaylist,
  removePlaylistRequest,
  removePlaylistSuccess,
  removePlaylistFailure,
} from '@client/store/playlist/actions';
import client from '@client/utils/client';
import { PlaylistInterface } from '@server/module/playlist/playlist.model';

export const fetchManyPlaylists = () => async (dispatch: Dispatch) => {
  dispatch(fetchPlaylistsRequest());
  try {
    const response = await client.get('/playlists');
    dispatch(fetchPlaylistsSuccess(response.data));
  } catch (err) {
    dispatch(fetchPlaylistsFailure());
  }
};

export const fetchSinglePlaylist = (id: number) => async (dispatch: Dispatch) => {
  dispatch(fetchPlaylistRequest(id));
  try {
    const response = await client.get(`/playlists/${id}`);
    dispatch(fetchPlaylistSuccess(response.data));
  } catch (err) {
    dispatch(fetchPlaylistFailure(id));
  }
};

export const setPlaylist = (playlist: PlaylistInterface) => (dispatch: Dispatch) => {
  dispatch(setSinglePlaylist(playlist));
};

export const removePlaylist = (id: number) => async (dispatch: Dispatch) => {
  dispatch(removePlaylistRequest(id));
  try {
    await client.delete(`/playlists/${id}`);
    dispatch(removePlaylistSuccess(id));
  } catch (err) {
    dispatch(removePlaylistFailure(id));
  }
};
