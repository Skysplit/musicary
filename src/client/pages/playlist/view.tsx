import { PureComponent, Fragment } from 'react';
import { bindActionCreators } from 'redux';
import { connect, Dispatch } from 'react-redux';
import { Request } from 'express';
import io from 'socket.io-client';
import { Snackbar } from 'react-md';
import { Context } from 'next/document';
import Error from 'next/error';
import { PlaylistInterface } from '@client/store/playlist';
import { setSinglePlaylist, removePlaylistTrack } from '@client/store/playlist/actions';
import client, { getHeaders } from '@client/utils/client';
import { fetchTracksSuccess, setSingleTrack, removeTrack } from '@client/store/track/actions';
import { TrackInterface } from '@client/store/track';
import { getUserToken } from '@client/utils/userData';
import PlaylistViewContainer from '@client/containers/PlaylistViewContainer';

interface PageContext extends Context {
  req: Request;
}

interface InitialProps {
  title: string;
  playlist?: PlaylistInterface;
  tracks?: TrackInterface[];
  errorCode?: number;
}

interface PlaylistViewPageProps extends InitialProps {
  setSinglePlaylist: typeof setSinglePlaylist;
  fetchTracksSuccess: typeof fetchTracksSuccess;
  setSingleTrack: typeof setSingleTrack;
  removePlaylistTrack: typeof removePlaylistTrack;
}

type ComponentProps = PlaylistViewPageProps & {};

interface Toast {
  text: string;
}

type ComponentState = {
  toasts: Toast[],
};

type AuthHeaders = ReturnType<typeof getHeaders>;

export class PlaylistViewPage extends PureComponent<ComponentProps, ComponentState> {
  private io: SocketIOClient.Socket;

  static async fetchPlaylist(id: number, headers: AuthHeaders) {
    const response = await client.get(`/playlists/${id}`, { headers });
    return response.data as PlaylistInterface;
  }

  static async fetchPlaylistTracks(id: number, headers: AuthHeaders) {
    const response = await client.get(`/playlists/${id}/tracks`, { headers });
    return response.data as TrackInterface[];
  }

  static async getInitialProps({ req, query }: PageContext): Promise<InitialProps> {
    let playlist: PlaylistInterface;
    let tracks: TrackInterface[];
    let errorCode: number;
    let title = 'Playlist';

    try {
      const headers = getHeaders(req);

      [playlist, tracks] = await Promise.all([
        PlaylistViewPage.fetchPlaylist(query.id as number, headers),
        PlaylistViewPage.fetchPlaylistTracks(query.id as number, headers),
      ]);

      title = `Playlist | ${playlist.name}`;
    } catch (err) {
      errorCode = err.response.status;
    }

    return {
      title,
      playlist,
      tracks,
      errorCode,
    };
  }

  state: ComponentState = {
    toasts: [],
  };

  constructor(props: ComponentProps) {
    super(props);

    if (props.playlist) {
      props.setSinglePlaylist(props.playlist);
      props.fetchTracksSuccess(props.playlist.id, props.tracks);
    }
  }

  private handleSocketConnection = () => {
    const { toasts } = this.state;
    const toast = { text: 'Connected with server!' };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketImportError = (playlistId: number) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: 'There was an error while importing playlist',
    };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTrack = (playlistId: number, track: TrackInterface) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `New track! ${track.name}`,
    };

    this.props.setSingleTrack(playlistId, track);

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTracksAccepted = (playlistId: number) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `Your tracks are being processed. They will appear on your list soon`,
    };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTrackRemoved = (playlistId: number, track: TrackInterface) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `Track ${track.name} removed from playlist`,
    };

    this.setState({
      toasts: [...toasts, toast],
    });

    this.props.removePlaylistTrack(playlistId, track.id);
  }

  componentDidMount() {
    this.io = io({
      query: { token: getUserToken() },
      autoConnect: false,
    });

    this.io.connect();

    this.io.on('connect', this.handleSocketConnection);
    this.io.on('tracks-accepted', this.handleSocketTracksAccepted);
    this.io.on('track', this.handleSocketTrack);
    this.io.on('import-error', this.handleSocketImportError);
    this.io.on('track-removed', this.handleSocketTrackRemoved);
  }

  componentWillUnmount() {
    this.io.removeAllListeners();
    this.io.close();
  }

  handleDismiss = () => {
    const [, ...toasts] = this.state.toasts;
    this.setState({ toasts });
  }

  render() {
    const { errorCode, playlist } = this.props;

    if (errorCode) {
      return <Error statusCode={errorCode} />;
    }

    return (
      <Fragment>
        <PlaylistViewContainer playlistId={playlist.id} />
        <Snackbar
          toasts={this.state.toasts}
          onDismiss={this.handleDismiss}
          autohideTimeout={2000}
          autohide
        />
      </Fragment>
    );
  }
}

const actionCreators = {
  setSinglePlaylist,
  setSingleTrack,
  removePlaylistTrack,
  fetchTracksSuccess,
};

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(actionCreators, dispatch);

export default connect(null, mapDispatchToProps)(PlaylistViewPage);
