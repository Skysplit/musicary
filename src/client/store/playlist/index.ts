import { omit, map, uniq, reject } from 'lodash';
import {
  fetchPlaylistRequest,
  fetchPlaylistSuccess,
  fetchPlaylistsRequest,
  fetchPlaylistsSuccess,
  fetchPlaylistFailure,
  fetchPlaylistsFailure,
  setSinglePlaylist,
  removePlaylistRequest,
  removePlaylistSuccess,
  removePlaylistFailure,
  removePlaylistTrack,
} from '@client/store/playlist/actions';
import {
  FETCH_SINGLE_PLAYLIST_REQUEST,
  FETCH_SINGLE_PLAYLIST_SUCCESS,
  FETCH_MANY_PLAYLISTS_REQUEST,
  FETCH_MANY_PLAYLISTS_SUCCESS,
  FETCH_MANY_PLAYLISTS_FAILURE,
  FETCH_SINGLE_PLAYLIST_FAILURE,
  SET_PLAYLIST,
  REMOVE_PLAYLIST_REQUEST,
  REMOVE_PLAYLIST_SUCCESS,
  REMOVE_PLAYLIST_FAILURE,
  REMOVE_PLAYLIST_TRACK,
} from '@client/store/playlist/actionTypes';
import { fetchTracksSuccess, setSingleTrack } from '@client/store/track/actions';
import { FETCH_MANY_TRACKS_SUCCESS, SET_TRACK } from '@client/store/track/actionTypes';

export interface PlaylistInterface {
  id?: number;
  name: string;
  position?: number;
  tracks?: number[];
}

export interface PlaylistState {
  isLoading: boolean;
  byId: {
    [key: number]: {
      isLoading: boolean;
      isRemoving: boolean;
      data?: PlaylistInterface;
    };
  };
}

type Actions =
  | ReturnType<typeof fetchPlaylistRequest>
  | ReturnType<typeof fetchPlaylistSuccess>
  | ReturnType<typeof fetchPlaylistFailure>
  | ReturnType<typeof fetchPlaylistsRequest>
  | ReturnType<typeof fetchPlaylistsSuccess>
  | ReturnType<typeof fetchPlaylistsFailure>
  | ReturnType<typeof setSinglePlaylist>
  | ReturnType<typeof removePlaylistRequest>
  | ReturnType<typeof removePlaylistSuccess>
  | ReturnType<typeof removePlaylistFailure>
  | ReturnType<typeof fetchTracksSuccess>
  | ReturnType<typeof setSingleTrack>
  | ReturnType<typeof removePlaylistTrack>;


const initialState: PlaylistState = {
  isLoading: false,
  byId: {},
};

export default (state: PlaylistState = initialState, action: Actions): PlaylistState => {
  switch (action.type) {
    case FETCH_MANY_PLAYLISTS_REQUEST: {
      return {
        ...state,
        isLoading: true,
      };
    }

    case FETCH_MANY_PLAYLISTS_SUCCESS: {
      return {
        ...state,
        isLoading: false,
        byId: action.payload.reduce(
          (accum, data) => Object.assign(accum, {
            [data.id]: { data, isLoading: false },
          }),
          {},
        ),
      };
    }

    case FETCH_MANY_PLAYLISTS_FAILURE: {
      return state;
    }

    case FETCH_SINGLE_PLAYLIST_REQUEST: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            isLoading: true,
            isRemoving: false,
          },
        },
      };
    }

    case FETCH_SINGLE_PLAYLIST_SUCCESS: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload.id]: {
            isLoading: false,
            isRemoving: false,
            data: action.payload,
          },
        },
      };
    }

    case FETCH_SINGLE_PLAYLIST_FAILURE: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            isLoading: false,
            isRemoving: false,
          },
        },
      };
    }

    case REMOVE_PLAYLIST_REQUEST: {
      const item = state.byId[action.payload];
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            ...item,
            isRemoving: true,
          },
        },
      };
    }

    case REMOVE_PLAYLIST_SUCCESS: {
      return {
        ...state,
        byId: omit(state.byId, [action.payload]),
      };
    }

    case REMOVE_PLAYLIST_FAILURE: {
      const item = state.byId[action.payload];
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            ...item,
            isRemoving: false,
          },
        },
      };
    }

    case SET_PLAYLIST: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload.id]: {
            isLoading: false,
            isRemoving: false,
            data: action.payload,

          },
        },
      };
    }

    case FETCH_MANY_TRACKS_SUCCESS: {
      const { playlistId, tracks } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return state;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: map(tracks, 'id'),
            },
          },
        },
      };
    }

    case SET_TRACK: {
      const { playlistId, track } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: uniq([
                ...playlist.data.tracks,
                track.id,
              ]),
            },
          },
        },
      };
    }

    case REMOVE_PLAYLIST_TRACK: {
      const { playlistId, trackId } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: reject(playlist.data.tracks, id => id === trackId),
            },
          },
        },
      };
    }

    default: {
      return state;
    }
  }
};
