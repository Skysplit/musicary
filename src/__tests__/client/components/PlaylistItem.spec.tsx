import React from 'react';
import { shallow } from 'enzyme';
import PlaylistItem from '@client/components/PlaylistItem';
import { PlaylistInterface } from '@client/store/playlist';

describe('<PlaylistItem />', () => {
  test('should render component', () => {
    const props = {
      playlist: { id: 1, name: 'playlistName' } as PlaylistInterface,
      onRemove: jest.fn(),
      onEdit: jest.fn(),
      removing: false,
    };

    const wrapper = shallow(<PlaylistItem {...props} />);
    const event = { stopPropagation: jest.fn() };

    window.confirm = jest.fn(() => true)

    wrapper.find('.edit').simulate('click', event);
    wrapper.find('.remove').simulate('click', event);

    expect(wrapper).toMatchSnapshot();
    expect(props.onEdit).toBeCalled();
    expect(props.onRemove).toBeCalled();
    expect(event.stopPropagation).toHaveBeenCalledTimes(2);
  });
});
