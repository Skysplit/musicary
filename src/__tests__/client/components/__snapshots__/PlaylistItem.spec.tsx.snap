// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaylistItem /> should render component 1`] = `
<LinkRoutes
  params={
    Object {
      "id": 1,
    }
  }
  route="playlist"
>
  <Grid
    component="div"
    gutter={0}
    noSpacing={true}
    spacing={0}
    stacked={false}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <Cell
      align="middle"
      component="div"
      desktopHidden={false}
      desktopSize={11}
      phoneHidden={false}
      phoneSize={3}
      tabletHidden={false}
      tabletSize={6}
    >
      <span
        className="md-title"
      >
        playlistName
      </span>
    </Cell>
    <Cell
      component="div"
      desktopHidden={false}
      desktopSize={1}
      phoneHidden={false}
      phoneSize={1}
      style={
        Object {
          "textAlign": "right",
        }
      }
      tabletHidden={false}
      tabletSize={2}
    >
      <withInk(withTooltip(Button))
        className="edit"
        disabled={false}
        icon={true}
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}
        onClick={[Function]}
        primary={true}
      >
        edit
      </withInk(withTooltip(Button))>
       
      <withInk(withTooltip(Button))
        className="remove"
        disabled={false}
        icon={true}
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}
        onClick={[Function]}
        secondary={true}
      >
        delete
      </withInk(withTooltip(Button))>
    </Cell>
  </Grid>
</LinkRoutes>
`;
